staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -4206,6 +4206,16 @@ type PodSpec struct {
 	// +featureGate=PodLevelResources
 	// +optional
 	Resources *ResourceRequirements `json:"resources,omitempty" protobuf:"bytes,40,opt,name=resources"`
+	// hostnameOverride specifies an explicit override for the Pod's hostname.
+	// When this field is set to a non-empty string:
+	// - It takes precedence over the values set in `hostname` and `subdomain`.
+	// - The Pod's hostname will be set to this value.
+	// - `setHostnameAsFQDN` must be nil or set to false.
+	// - `hostNetwork` must be set to false.
+	//
+	// This field must be a valid DNS subdomain as defined in RFC 1123 and contain at most 64 characters.
+	// Requires the HostnameOverride feature gate to be enabled.
```

Comment:
Maybe it should be like this?

```suggestion
	// Requires the HostnameOverride feature gate to be enabled.
	//
	// +featureGate=HostnameOverride
	// +optional
```